### ISSUE TITLE
[scaffolding-chef-inspec] Accept license during build?

### ISSUE THREAD

**Issue by jerryaldrichiii** [chef/effortless#107]
Hey all, in `chef/scaffolding-chef-inspec` `inspec archive` is called during build and requires that the license be accepted. 

Currently in order to do this you must do `CHEF_LICENSE='accept-no-persist' hab pkg build .`. 

What are your thoughts on a better UX for handing this? Maybe an error in `do_build` if the env var isn’t set with a descriptive error? Accept the license for the user for `inspec archive` only? Another habitat-esque way to pass in the license acceptance (maybe via default.toml?)

**Comment by jerryaldrichiii** [comment]
I lied, the UX is a bit worse. You have to know that `HAB_` env vars get passed to studio so you have to set `HAB_CHEF_LICENSE`.

**Comment by jerryaldrichiii** [comment]
Actually I lied, the UX is worse. I couldn't make either of the following work:
  - `CHEF_LICENSE='accept-no-persist' hab pkg build .`
  - `HAB_STUDIO_SECRET_CHEF_LICENSE='accept-no-persist' hab pkg build .`

Instead I had to `hab studio enter` then `CHEF_LICENSE='accept-no-persist' hab pkg build .`

This prevents pipelines from doing build I suspect...unless I'm missing something.

**Comment by TheLunaticScripter** [comment]
@jerryaldrichiii The Habitat way to set this is with a `.studiorc` file that contains `export CHEF_LICENSE=accept-no-persist` we cannot legally accept the Chef License for the user base in the scaffolding and this environment variable is the standard across Chef Products. I'll keep this issue open so we can be sure to add something to the README for this.

**Comment by jerryaldrichiii** [comment]
@TheLunaticScripter does the `.studiorc` run when not doing `hab studio enter`? I can't make it work with:
  - `hab pkg build .`
  - `hab studior run 'build'`

**Comment by TheLunaticScripter** [comment]
Propose adding InSpec license to plan file as a variable.

**Comment by smford22** [comment]
You can accept the license in the following way and build jobs will succeed

```
pkg_name=audit-baseline
pkg_version=0.1.0
pkg_origin=<YOUR ORIGIN>
pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
pkg_license=("Apache-2.0")
pkg_description="Effortless Linux Audit Example"
pkg_scaffolding="chef/scaffolding-chef-inspec"

do_setup_environment() {
 set_runtime_env CHEF_LICENSE "accept-no-persist"
}
```

**Comment by robbkidd** [comment]
### Proposal

We update the scaffolds so that `accept-no-persist` is set for build-time activities.

### Reasoning

We _know_ that `chef install` and `inspec archive` are being used for automated build purposes at their call sites, not for runtime use. So long as the artifact produced still requires active licenses acceptance at _its_ runtime, we can hard-code `accept-no-persist` scoped to the actions taken in the build environment.

### Possible Implementation

To cover the usage of our tooling as built-time actions:

* Add the following to `do_default_prepare`:

```
export CHEF_LICENSE="accept-no-persist"
```

* Add the following to `Invoke-DefaultPrepare`:

```
$env:CHEF_LICENSE="accept-no-persist"
```
---

I do not recommend this:

```
do_setup_environment() {
 set_runtime_env CHEF_LICENSE "accept-no-persist"
}
```

Despite "no-persist" in the value, this environment variable is now set—and in effect, persisted—within the plan and package for build _and_ runtime. The consumers of the package built by this plan are not required to actively accept the license in the runtime environments.

**Comment by TheLunaticScripter** [comment]
This is fixed in 0.15.0

**Comment by aymendr** [comment]
Try to run **hab studio enter** with sudo user, this worked for me and licence acceptance is then prompted
